bigchaindb/pipelines/block.py

```diff
@@ -73,10 +74,18 @@ def validate_tx(self, tx):
 
         # If transaction is not valid it should not be included
         try:
+            # Do not allow an externally submitted GENESIS transaction.
+            # A simple check is enough as a pipeline is started only after the
+            # creation of GENESIS block, or after the verification of a GENESIS
+            # block. Voting will fail at a later stage if the GENESIS block is
+            # absent.
+            if tx.operation == Transaction.GENESIS:
+                raise GenesisBlockAlreadyExistsError('Duplicate GENESIS transaction')
+
             tx.validate(self.bigchain)
             return tx
         except ValidationError as e:
-            logger.warning('Invalid tx: %s', e)
+            logger.error('Invalid tx: %s', e)
```

Comment:
Maybe I am just being picky here but I don't think this should be treated as an error since no error as actually occurred. An invalid transaction was submitted and was correctly handled by the pipeline.